File: es2_0_test2/src/main/java/com/opengles/book/es2_0_test2/muti/surface/MyGLSurfaceView.java
package com.opengles.book.es2_0_test2.muti.surface;

import android.content.Context;
import android.util.AttributeSet;

import com.opengles.book.es2_0_test2.eglUtils.EglSurfaceView;

/**
 * author: ycl
 * date: 2019-01-04 10:56
 * desc:
 */
public class MyGLSurfaceView extends EglSurfaceView {
    public MyGLSurfaceView(Context context) {
        this(context,null);
    }

    public MyGLSurfaceView(Context context, AttributeSet attrs) {
        this(context, attrs,0);
    }

    public MyGLSurfaceView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        setRenderer(new MyRender(context));
        setRenderMode(EglSurfaceView.RENDERMODE_WHEN_DIRTY);
    }
}

File: es2_0/src/main/java/com/opengles/book/es2_0/base/MyGlSurfaceView.kt
package com.opengles.book.es2_0.base

import android.content.Context
import android.opengl.GLSurfaceView
import android.util.AttributeSet

/**
 * author: ycl
 * date: 2018-10-28 14:44
 * desc:
 */
class MyGlSurfaceView : GLSurfaceView {
    private var mMyRenderer: MyRenderer? = null

    constructor(context: Context?) : this(context, null)
    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs) {
        init()
    }

    private fun init() {
        setEGLContextClientVersion(2)// 使用gl2.0
        mMyRenderer = MyRenderer(this@MyGlSurfaceView)
        setRenderer(mMyRenderer)
        renderMode = GLSurfaceView.RENDERMODE_WHEN_DIRTY // 渲染一次
    }

    fun setRendererClass(rendererClass: Class<out BaseRenderer>) {
        try {
            mMyRenderer?.setRendererClass(rendererClass)
        } catch (e: Exception) {
            e.printStackTrace()
        }
        requestRender()
    }

}